internal/gatewayapi/securitypolicy.go

```diff
@@ -494,6 +495,12 @@ func (t *Translator) buildOIDC(
 		logoutPath = *oidc.LogoutPath
 	}
 
+	h := fnv.New32a()
```

Comment:
can we reuse https://github.com/envoyproxy/gateway/blob/4c79ef9aa1e06d3b61b83e95f21d1705fa6568ba/internal/provider/utils/utils.go#L39 ?

Reply:
`HashString` uses SHA256, which is theoretically slower than FNV-32, and generates a 256-bit result.

I don't think it matters that much for this purpose that the hash being used is slower, or crypto-secure, but using the full 256 bits in the cookie suffix would result in 64 characters instead of the current 8. I think that would be a little long for a cookie name. 

Using only 32 bit from the result would theoretically be OK for this purpose, but I would prefer to use the FNV-32 hash function here so that the entire hash result would still be used.

Reply:
move it into a more common package, e.g. `internal/util`

Reply:
non blocking comment - less code, lesser bugs, lesser vulnerabilities, if this hash function is better, lets replace it with the one in `provider/utils`, as @zirain suggests, if `provider/utils` is the wrong home, lets move it to `utils`

Reply:
Will do in a follow-up PR because it will impact many places.

Reply:
https://github.com/envoyproxy/gateway/pull/2681